### PR TITLE
Make thymeleaf context relative url resolution aware of context-path fixes #347

### LINKED ISSUE #347: Thymeleaf context relative url resolution is ignoring the context-path property
### Expected Behavior

Starting my application with the `context-path: myapp`, and having a view with `<a th:href="@{/page1/}">Page 1</a>`, I expect the final url to be `<a href="/myapp/page1/">Page 1</a>`.

### Actual Behaviour

The context path is completely ignored, i.e. `<a href="/page1/">Page 1</a>`

### Steps To Reproduce

- Create Micronaut Thymeleaf project
- Set the `context-path: myapp`
- Write context relative link expression <a th:href="@{/page1/}">Page 1</a> in a template file and use it with `/index`
- Add the `/page1` 
- Run the app and click the link on `/myapp/index`
- Not found

**Or clone [this repo](https://github.com/Iman78/thymeleaf-context-path), start the app and go to `/myapp/views`, click the link in there.**

### Environment Information

- Darwin
- JDK 11

### Example Application

https://github.com/Iman78/thymeleaf-context-path

### Version

3.2.6

### PR DESCRIPTION
**Pull request by Iman78**

### CODE REVIEW

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java [hunk 1]
```diff
@@ -32,6 +33,12 @@
 @Singleton
 public class LinkBuilder extends StandardLinkBuilder {
 
+    private final String contextPath;
+
+    public LinkBuilder(@Nullable @Property(name = "micronaut.server.context-path") String contextPath) {
```
Comment (graemerocher) [inline views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java:38]: This is a breaking change, could you restore the original constructor and annotate this one with `@Inject`

A test would be nice too

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java [hunk 2]
```diff
@@ -32,6 +34,13 @@
 @Singleton
 public class LinkBuilder extends StandardLinkBuilder {
 
+    @Inject
+    @Nullable
+    @Property(name = "micronaut.server.context-path")
+    private String contextPath;
```
Comment (graemerocher) [inline views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java:40]: Could you use constructor injection, private field injection is a bit of an anti-pattern.

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java (hunk 2 above)
Comment (Iman78) [reply views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java:40]: In that case there would still be code somewhere using this class without context path, or did I miss something 😅

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java [hunk 3]
```diff
@@ -32,6 +34,16 @@
...
 
+    @Nullable
+    private String contextPath;
+
+    @Inject
+    public LinkBuilder(@Nullable @Property(name = "micronaut.server.context-path") String contextPath) {
+        this.contextPath = contextPath;
+    }
+
+    public LinkBuilder() { }
```
Comment (sdelamo) [inline views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java:45]: annotate old one with @Depreacated and tell to use the other controller

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java (hunk 3 above)
Comment (sdelamo) [reply views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/LinkBuilder.java:53]: resolved via https://github.com/micronaut-projects/micronaut-views/pull/348/commits/08005a42c634addc476418d105594c618a60ac11

**Comment by sdelamo** [comment]
[I've injected `HttpServerConfiguration` instead of `@Property(name = "micronaut.server.context-path"](https://github.com/micronaut-projects/micronaut-views/pull/348/commits/c8ca9f6379dae84d3f698ec4a7cf368365c6d0ab) which I think it is cleaner.

**Review by sdelamo — APPROVED** [review]
Thanks for your contribution @Iman78

**Pull request merged**